src/services/WorkPeriodPaymentService.js

```diff
@@ -55,47 +56,29 @@ async function _createSingleWorkPeriodPayment (workPeriodPayment, createdBy) {
  * @returns {Object} the created workPeriodPayment
  */
 async function _createSingleWorkPeriodPaymentWithWorkPeriodAndResourceBooking (workPeriodPayment, createdBy, correspondingWorkPeriod, correspondingResourceBooking) {
-  if (!correspondingResourceBooking.billingAccountId) {
-    throw new errors.ConflictError(`id: ${correspondingWorkPeriod.resourceBookingId} "ResourceBooking" Billing account is not assigned to the resource booking`)
+  if (_.isNil(correspondingResourceBooking.billingAccountId)) {
+    throw new errors.ConflictError(`id: ${correspondingResourceBooking.id} "ResourceBooking" Billing account is not assigned to the resource booking`)
   }
   workPeriodPayment.billingAccountId = correspondingResourceBooking.billingAccountId
+  if (_.isNil(correspondingResourceBooking.memberRate)) {
+    throw new errors.ConflictError(`Can't find a member rate in ResourceBooking: ${correspondingResourceBooking.id} to calculate the amount`)
+  }
```

Comment:
We should also return an error if `memberRate` is `0`.

Reply:
@eisbilir also here is small issue. General idea, we should not allow creating payments with 0 amount.